src/openLibrary/producer/parser.js

```diff
@@ -0,0 +1,402 @@
+ */
+
+
+import {entityTypes, isNotDefined, sortName} from '../../helpers/utils';
+import {identifiers, mapLanguage} from '../../helpers/mapping';
+import _ from 'lodash';
+import franc from 'franc-min';
+
+
+const WORK = 'work';
+const EDITION = 'edition';
+const AUTHOR = 'author';
+
+
+function detectLanguage(name) {
+	let lang = franc(name);
+	lang = lang !== 'und' ? lang : 'eng';
+
+	return mapLanguage(lang);
+}
+
+function processWork(json) {
+	if (isNotDefined(json)) {
+		return null;
+	}
+
+	const work = {};
+	work.alias = [];
+	work.metadata = {};
+	work.metadata.relationships = [];
+	work.metadata.links = [];
+	work.identifiers = [];
+
+	work.entityType = entityTypes.WORK;
+	work.source = 'OPENLIBRARY';
+
+	// Set up aliases
+	// The first alias (in the logical flow) is made to be default alias
+	let defaultAlias = true;
+	if (!isNotDefined(json.title)) {
+		const name = json.title;
+		const lang = detectLanguage(name);
+
+		work.alias.push({
+			default: defaultAlias,
+			languageId: lang,
+			name,
+			primary: true,
+			sortName: sortName(name)
+		});
+		defaultAlias = false;
+	}
+	if (!isNotDefined(json.subtitle)) {
+		const name = json.subtitle;
+		const lang = detectLanguage(name);
+
+		work.alias.push({
+			default: defaultAlias,
+			languageId: lang,
+			name,
+			sortName: sortName(name)
+		});
+		defaultAlias = false;
+	}
+
+
+	// Last Edited at source
+	if (!isNotDefined(_.get(json, 'last_modified.value'))) {
+		work.lastEdited = json.last_modified.value;
+	}
+
+	if (!isNotDefined(json.key)) {
+		const openLibraryWorkId = json.key.split('/')[2];
+		work.identifiers.push({
+			typeId: identifiers.openLibraryWorkId,
+			value: openLibraryWorkId
+		});
+		work.originId = openLibraryWorkId;
+	}
+
+	if (!isNotDefined(json.authors) && (json.authors instanceof Array)) {
+		json.authors.forEach((authorObj) => {
+			if (!isNotDefined(authorObj && _.get(authorObj, 'author.key'))) {
+				work.metadata.relationships.push({
+					type: 'authoredBy',
+					value: authorObj.author.key.split('/')[2]
+				});
+			}
+		});
+	}
+
+	if (!isNotDefined(json.description)) {
+		work.disambiguation = json.description.value;
+	}
+
+	if (!isNotDefined(json.links) && (json.links instanceof Array)) {
+		json.links.forEach(link => {
+			if (!isNotDefined(link.title && link.url)) {
+				work.metadata.links.push({
+					title: link.title,
+					url: link.url
+				});
+			}
+		});
+	}
+
+	const metadataFields = [
+		'subjects',
+		'subject_places',
+		'subject_people',
+		'description',
+		'subject_times',
+		'cover_edition',
+		'works',
+		'lc_classifications',
+		'first_publish_date',
+		'dewey_number',
+		'first_sentence',
+		'excerpts',
+		'number_of_editions',
+		'remote_ids'
+	];
+
+	metadataFields.forEach(field => {
+		if (!isNotDefined(json[field])) {
+			work.metadata[field] = json[field];
+		}
+	});
+
+	return work;
+}
+
+function processAuthor(json) {
+	if (isNotDefined(json)) {
+		return null;
+	}
+
+	const creator = {};
```

Comment:
Same comment as for work - just build the object in the initial assignment.

Reply:
Indeed. This actually is a very sloppy work. I have some uncommitted incomplete code - which I will clean up and commit soon. It's not really hampering the process - and is more of simply refactoring - so I guess it can wait.